pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1254,7 +1254,8 @@ private CompletableFuture<Optional<Topic>> createNonPersistentTopic(String topic
             nonPersistentTopic = newTopic(topic, null, this, NonPersistentTopic.class);
         } catch (Throwable e) {
             log.warn("Failed to create topic {}", topic, e);
-            return FutureUtil.failedFuture(e);
+            topicFuture.completeExceptionally(e);
```

Comment:
Makes sense to me, I also tried to fix the flaky test, and during the process, I find that I didn't handle this case. 
But the test keeps failing after fixed, so the issue never get fixed....
https://github.com/apache/pulsar/pull/22256/files#diff-0210356c8a88e4efa89eb769a027fa6c166db479dbad8bbbbc704c6ed6e317f5R1254